searx/search/__init__.py

```diff
@@ -37,6 +39,38 @@ def initialize(settings_engines=None, enable_checker=False, check_network=False,
         initialize_checker()
 
 
+def replace_auto_language(search_query: SearchQuery):
+    """
+    The `language identification model`_ is harmonized with the SearXNG's language
+    (locale) model.  General conditions of SearXNG's locale model are:
+    a. SearXNG's locale of a query is passed to the
+    :py:obj:`searx.locales.get_engine_locale` to get a language and/or region
+    code that is used by an engine.
+    b. SearXNG and most of the engines do not support all the languages from
+    language model and there might be also a discrepancy in the ISO-639-3 and
+    ISO-639-2 handling (:py:obj:`searx.locales.get_engine_locale`).  Further
+    more, in SearXNG the locales like ``zh-TH`` (``zh-CN``) are mapped to
+    ``zh_Hant`` (``zh_Hans``).
+    Conclusion: This function  does only auto-detect the languages a user can select
+    in the language menu (:py:obj:`supported_langs`).
+    .. hint::
+    """
```

Comment:
This function description confuses me .. its partial C&P of the plugin description but the markup is broken and the content is different to what the function does.

By example: *is harmonized with the SearXNG's language (locale) model.* --> IMO it isn't .. in the plugin we had 

```python
    if lang in supported_langs:
        search.search_query.lang = lang
        try:
            search.search_query.locale = babel.Locale.parse(lang)
        except babel.core.UnknownLocaleError:
            pass
```

and now we have no longer `if lang in supported_langs`.

Reply:
Updated: 
* the signature of `searx.utils.detect_language` is now:
```python
def detect_language(text: str, threshold: float = 0.3, only_search_languages: bool = False) -> Optional[str]:
```
* `searx/search/__init__.py` sets `only_search_languages` to `True`.